kie-server-parent/kie-server-services/kie-server-services-optaplanner/src/main/java/org/kie/server/services/optaplanner/OptaplannerKieServerExtension.java

```diff
@@ -82,17 +82,16 @@ public void init(KieServerImpl kieServer, KieServerRegistry registry) {
         this.registry = registry;
         // the following threadpool will have a max thread count equal to the number of cores on the machine.
         // if new jobs are submited and all threads are busy, the reject policy will kick in.
-        int poolSize = Runtime.getRuntime().availableProcessors();
-        if ( poolSize >= 4 ) {
-            // Leave 1 processor alone to handle REST/JMS requests and run the OS
-            poolSize--;
-        }
+        int availableProcessorCount = Runtime.getRuntime().availableProcessors();     
+        int resolvedActiveThreadCount = availableProcessorCount <= 2 ? 1 : availableProcessorCount - 2;
```

Comment:
Could instead write this as `Math.max(1, availableProcessorCount - 2)`. That is probably a bit easier to read.

Reply:
Doing this change in the optaplanner-core code that does something similar too.

Reply:
When it comes to soft things (such as code style, etc), we are more lenient with contributors than we are to full-time committers who have the time to make sure all the i's are dotted and all t's are crossed. We'll take care of dotting and crossing those.
Thanks for the contribution :)